content/guest-post/secure-php-api-using-jwt/index.md

```diff
@@ -0,0 +1,254 @@
+    },
+    body: JSON.stringify({
+      username: formData.inputEmail.value,
+      password: formData.inputPassword.value
+    })
+  });
+
+  if (response.status >= 200 && response.status <= 299) {
+    const jwt = await response.text();
+    storeJWT.setJWT(jwt);
+    frmLogin.style.display = 'none';
+    btnResource.style.display = 'block';
+  } else {
+    // Handle errors
+    console.log(response.status, response.statusText);
+  }
+});
+```
+You've already produced and submitted the JWT to the user. So now it's time to put the JWT to use on the user side.
+
+### Using JWT
+
+As previously stated, if the form submission is successful, you'll display a button to obtain a timestamp. This button will invoke the GET method on the `resource.php` script. The `resource.php` script will then set the JWT obtained after successful authentication in the authentication header.
+
+The following code will help you achieve this feat.
+
+```javascript
+btnResource.addEventListener('click', async (e) => {
+  const result = await fetch('/resource.php', {
+    headers: {
+      'Authorization': `Bearer ${storeJWT.JWT}`
+    }
+  });
+  const timeStamp = await result.text();
+  console.log(timeStamp);
+});
+```
+Run the program once you've written this code and enter your credentials into the form fields. A GET request will be sent when you click the submit button. Here is a sample GET request to assist you in making the right identification.
+
+```php
+GET /resource.php HTTP/1.1
+Host: yourhost.com
+Connection: keep-alive
+Accept: */*
+X-Requested-With: XMLHttpRequest
+Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvcWEtYXBpLndlbGx2aWJlLmNvbVwvYXBpXC9hdXRoXC9sb2dpbiIsImlhdCI6MTYzMDQ3OTA5NSwiZXhwIjoxNjMwNDgyNjk1LCJuYmYiOjE2MzA0NzkwOTUsImp0aSI6Imtsa3hHUGpMOVlNTzRSdUsiLCJzdWIiOjc3ODE4LCJwcnYiOiIyM2JkNWM4OTQ5ZjYwMGFkYjM5ZTcwMWM0MDA4NzJkYjdhNTk3NmY3IiwidXNlcnNfaWQiOjc3ODE4LCJtZW1iZXJzX2lkIjo3Nzg4MzMsInByb3h5X3VzZXJfbWVtYmVyc19pZCI6bnVsbH0.TxXwLLu1zWBe7cLLYdFYy3P2HX4AaLgc7WfSRtTgeiI
+```
+
+### Validating JWT
+If you've followed along until here, everything should be working fine. Now, let's begin with validating the JWT.
+
+To assist us with this operation, you've previously added the composer's autoloader function. You'll now use the `preg match` function to extract the token from the Bearer header. For this extraction, you'll use the `preg match` function and supply a regular expression.
+
+```php
+if (! preg_match('/Bearer\s(\S+)/', $_SERVER['HTTP_AUTHORIZATION'], $matches)) {
+    header('HTTP/1.0 400 Bad Request');
+    echo 'Token not found in request';
+    exit;
+}
+```
+
+The code above attempts to extract the token from the Bearer header. In this case, error handling is utilized. Thus if the token is not discovered, an HTTP 404 error is displayed to the user.
+
+To validate the JWT, you must first compare it to the previously created JWT.
+
+```php
+$jwt = $matches[1];
```

Comment:
Please confirm if it should be `$matches[1]` or `$matches[0]`. If there is no change here, change `first index` part of line 224 accordingly.